src/App.ts

```diff
@@ -7,83 +7,193 @@ import * as bodyParser from "body-parser";
   );
   app.post(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_ACTIVATION,
-    (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Activation Request (POST)...");
-      toExpressHandler(
-        PaymentController.activatePaymentToPagoPa(
-          config.PAGOPA,
-          attivaRPTPagoPaClient
-        )
-      )(req, res);
-    }
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_POST,
+    jsonParser,
+    activatePaymentHandler
+  );
+
+  const getActivationStatusHandler = toExpressHandler(
+    PaymentController.getActivationStatus(redisClient)
   );
+  app.get(
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_GET,
+    getActivationStatusHandler
+  );
+
+  // Endpoint for OpenAPI handler
+  app.get("/specs/api/v1/swagger.json", GetOpenapi(publicApiV1Specs));
+}
+
+/**
+ * Define a redis client necessary to handle persistent data
+ * @param {Configuration} config - Server Configuration
+ * @return {(app: core.Express) => soap.Server} A method to execute for start server listening
+ */
+function getRedisClient(config: Configuration): redis.RedisClient {
+  if (!config.REDIS_DB.USE_CLUSTER) {
+    logger.debug("Creating a REDIS client...");
+    return redis.createClient(config.REDIS_DB.HOST);
+  }
+  logger.debug("Creating a REDIS client using cluster...");
+  return new RedisClustr({
+    redisOptions: {
+      auth_pass: config.REDIS_DB.PASSWORD,
+      tls: {
+        servername: config.REDIS_DB.HOST
+      }
+    },
+    servers: [
+      {
+        host: config.REDIS_DB.HOST,
+        port: config.REDIS_DB.PORT
+      }
+    ]
+  }) as redis.RedisClient;
+  return redis.createClient(config.REDIS_DB.PORT);
+}
+
+/**
+ * Define a Service Handler for FespCdService SOAP service
+ * It's an endpoint for PagoPA to confirm a payment activation result
+ * @param {redis.RedisClient} redisClient - Redis Client to store persistente paymentId into DB
+ * @param {number} redisTimeoutSecs - Timeout set for information stored into DB
+ * @return {IFespCdPortTypeSoap} An object containing the service handler
+ */
+function getFespCdServiceHandler(
+  redisClient: redis.RedisClient,
+  redisTimeoutSecs: number
+): IFespCdPortTypeSoap {
+  return {
+    cdInfoPagamento: (
+      iCdInfoPagamentoInput: IcdInfoPagamentoInput,
+      callback: (
+        err: any | null, // tslint:disable-line:no-any
+        iCdInfoPagamentoOutput: IcdInfoPagamentoOutput,
+        raw: string,
+        soapHeader: { readonly [k: string]: any } // tslint:disable-line:no-any
+      ) => void
+    ) => {
+      PaymentController.setActivationStatus(
+        iCdInfoPagamentoInput,
+        redisTimeoutSecs,
+        redisClient
+      ).then(
+        iCdInfoPagamentoOutput => {
+          callback(undefined, iCdInfoPagamentoOutput, "", {});
+        },
+        err => {
+          logger.debug(`Error on setActivationStatus: ${err}`);
+          callback(
+            undefined,
+            {
```

Comment:
I believe you must return an object with a `Fault` key, [like this one](https://github.com/vpulim/node-soap#soap-fault)

Reply:
I didn't get this point. My method will return
{ esito: Esito.OK } or { esito: Esito.KO }
so I'll never need to handle a fault exception.

Where do you suggest to throw that exception?

Reply:
What I mean is that, since this is an internal server error, it may make sense to return a [SOAP fault](https://github.com/teamdigitale/italia-pagopa-api/blob/3617716bd30ca4e6553a6d297cea4699d9ed8525/wsdl-lib/FespCdService/FespCdPortType.ts#L16) instead a success response (albeit with Esito = KO) - I guess we need to understand what's the behavior on the pagoPA side to decide.

The error originates from Redis and it will probably be a transient error, so we want the PagoPA Nodo to retry. To decide whether to return a `Fault` or an `Esito.KO` we need to know which one will be considered as a transient error by the Nodo and trigger a retry. Makes sense?

Reply:
Yes, now I got your observation.

Actually, the protocol shared by SIA establishes just OK\KO responses and doesn't expect a fault. I'm just following the protocol defined by them.
It could be useful to talk about it with them. I created a story: #158759398